src/components/side_panel/chart/chart_with_axis/design_panel.ts

```diff
@@ -131,4 +143,74 @@ export class ChartWithAxisDesignPanel extends Component<Props, SpreadsheetChildE
     const dataSets = this.props.definition.dataSets;
     return dataSets[this.state.index]?.label || this.getDataSeries()[this.state.index];
   }
+
+  updateDataTrend(display: boolean) {
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return;
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      trend: {
+        type: "polynomial",
+        order: 3,
+        ...dataSets[this.state.index].trend,
+        display,
+      },
+    };
+    this.props.updateChart(this.props.figureId, { dataSets });
+  }
+
+  getDataTrend() {
+    const dataSets = this.props.definition.dataSets;
+    return dataSets?.[this.state.index]?.trend;
+  }
+
+  onChangeTrendType(ev: InputEvent) {
+    const type = (ev.target as HTMLInputElement).value as TrendType;
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return;
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      trend: {
+        ...dataSets[this.state.index].trend,
+        type,
+      },
+    };
+    this.props.updateChart(this.props.figureId, { dataSets });
+  }
+
+  onChangePolynomialDegree(ev: InputEvent) {
+    const element = ev.target as HTMLInputElement;
+    const order = parseInt((ev.target as HTMLInputElement).value);
+    if (order < 1) {
+      element.value = `${this.getDataTrend()?.order ?? 1}`;
+      return;
+    }
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return;
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      trend: {
+        ...dataSets[this.state.index].trend,
+        order,
+      },
+    };
+    this.props.updateChart(this.props.figureId, { dataSets });
+  }
+
+  getTrendLineColor() {
+    return this.getDataTrend()?.color ?? setColorAlpha(this.getDataSerieColor(), 0.5);
+  }
+
+  updateTrendLineColor(color: Color) {
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) return;
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      trend: {
+        ...dataSets[this.state.index].trend,
+        color,
+      },
+    };
+    this.props.updateChart(this.props.figureId, { dataSets });
```

Comment:
This part seems repeated 4 time in the panel. Maybe a method `updateTrendValue(key, value)` would help ?